kafka/src/main/java/com/opentable/logging/KafkaAppender.java

```diff
@@ -86,9 +86,7 @@ protected void append(ILoggingEvent eventObject)
             throw Throwables.propagate(e);
         }
 
-        final byte[] key = partitionShufflerKey.get();
-        key[0]++;
-        producer.send(new KeyedMessage<>(topic, key, out.toByteArray()));
+        producer.send(new KeyedMessage<>(topic, keyGenerator.next(), out.toByteArray()));
```

Comment:
How does this producer interface work?  You give it a byte array, and it hashes it to determine into which partition to put the item?  If so, then won't this implementation mean that different threads will be passing in the same keys since they're independently sequential?  If this is a problem, maybe namespace the keys by including an extra byte that indicates from which thread it came?